FWIO/RNTuple/plugins/RNTupleOutputModule.cc

```diff
@@ -0,0 +1,262 @@
+        compressionLevel_(pset.getUntrackedParameter<unsigned int>("compressionLevel")),
+        approxZippedClusterSize_(pset.getUntrackedParameter<unsigned long long>("approxZippedClusterSize")),
+        maxUnzippedClusterSize_(pset.getUntrackedParameter<unsigned long long>("maxUnzippedClusterSize")),
+        initialUnzippedPageSize_(pset.getUntrackedParameter<unsigned long long>("initialUnzippedPageSize")),
+        maxUnzippedPageSize_(pset.getUntrackedParameter<unsigned long long>("maxUnzippedPageSize")),
+        pageBufferBudget_(pset.getUntrackedParameter<unsigned long long>("pageBufferBudget")),
+        useBufferedWrite_(pset.getUntrackedParameter<bool>("useBufferedWrite")),
+        useDirectIO_(pset.getUntrackedParameter<bool>("useDirectIO")),
+        dropMetaData_(pset.getUntrackedParameter<bool>("dropPerEventDataProductProvenance")),
+        useStreamer_(pset.getUntrackedParameter<bool>("useStreamer")) {}
+
+  void RNTupleOutputModule::openFile(FileBlock const& fb) {
+    RNTupleOutputFile::Config conf;
+    conf.wantAllEvents = wantAllEvents();
+    conf.selectorConfig = selectorConfig();
+    conf.compressionAlgo = compressionAlgo_;
+    conf.compressionLevel = compressionLevel_;
+    conf.approxZippedClusterSize = approxZippedClusterSize_;
+    conf.maxUnzippedClusterSize = maxUnzippedClusterSize_;
+    conf.initialUnzippedPageSize = initialUnzippedPageSize_;
+    conf.maxUnzippedPageSize = maxUnzippedPageSize_;
+    conf.pageBufferBudget = pageBufferBudget_;
+    conf.useBufferedWrite = useBufferedWrite_;
+    conf.useDirectIO = useDirectIO_;
+    conf.dropMetaData = dropMetaData_;
+    conf.doNotSplitSubFields = noSplitSubFields_;
+    if (useStreamer_ and overrideStreamer_.empty()) {
+      auto const& prods = keptProducts()[InEvent];
+      conf.streamerProduct = std::vector<bool>(prods.size(), true);
+    } else if (not overrideStreamer_.empty()) {
+      auto const& prods = keptProducts()[InEvent];
+      conf.streamerProduct = std::vector<bool>(prods.size(), useStreamer_);
+      unsigned int index = 0;
+      for (auto const& prod : prods) {
+        auto choice = useStreamer(prod.first->branchName(), overrideStreamer_);
+        if (choice) {
+          std::cout << "found match " << prod.first->branchName() << std::endl;
+          if (*choice != useStreamer_) {
+            conf.streamerProduct[index] = *choice;
+          }
+        }
+        ++index;
+      }
+    }
+    assert(reg_);
+    file_ = std::make_unique<RNTupleOutputFile>(fileName_, fb, keptProducts(), conf, reg_->anyProductProduced());
+  }
+
+  void RNTupleOutputModule::initialRegistry(edm::ProductRegistry const& iReg) {
+    reg_ = std::make_unique<ProductRegistry>(iReg.productList());
+  }
+
+  void RNTupleOutputModule::reallyCloseFile() {
+    if (file_) {
+      assert(reg_);
+      file_->reallyCloseFile(*branchIDLists(), *thinnedAssociationsHelper(), *reg_);
+    }
+  }
+
+  RNTupleOutputModule::~RNTupleOutputModule() {}
```

Comment:
```suggestion
  RNTupleOutputModule::~RNTupleOutputModule() = default;
```
(or remove the explicit declaration+definition)